scripts/mac_build

```diff
@@ -0,0 +1,246 @@
+#!/bin/bash
```

Comment:
Can you update https://github.com/KratosMultiphysics/Kratos/blob/master/scripts/standard_configure_mac.sh?

Reply:
Not really, it has a fundamentally different assumption about the host system than `mac_build` :/

`standard_configure_mac.sh` is hard-coded to use the system `clang` and `python`, and requires the user to provide `boost`. I don't know whether these were deliberate decisions, so I don't want to change them.

`mac_build` uses `clang` provided by `llvm` separately installed from `Homebrew`. `boost` is also pulled in from `Homebrew`, and the environment `python` is used.

Essentially, the main difference is that `standard_configure_mac.sh` requires the user to modify their system packages and know what they're doing, while `mac_build` provides a specific way of installing dependencies without touching system packages.

Reply:
> You are free to change anything related to mac, as I have no clue on the environment of iOs or how packages are managed. If your solution is more convenient for the final user go ahead

I'm new to macos, so I've no idea how conventient this is for the average/veteran user, but I managed to install Kratos on 2 different kinds of hardware so far, so I'm quite happy with it.

I'll update our `standard_configure_mac.sh` once I'm more confident about what I'm doing.